### PR TITLE
Improve color extraction for media control card

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Improve our color extraction algorithm:

- Use most populous color as the background color (old one could filter it out)
- Take the 2nd most populous color. If that has correct contrast ratio, ship it. Otherwise we go 

Also includes Zac change to bump the contrast ratio from 3.5 to the correct 4.5. 

And lazy load the media control card as it includes node-vibrant. 

Before:
image

After:
image

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by balloob** [comment]
Left old, right new:

image

image

image

image

image

image

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 1]
```diff
@@ -59,6 +59,103 @@ function getContrastRatio(
...
+          logColor(compareColor, "PICKED");
+        }
+        foregroundColor = compareColor.hex;
+        break;
+      }
+    }
+  }
+
+  if (foregroundColor === undefined) {
+    foregroundColor = colors[0].bodyTextColor;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: For clarity:
```suggestion
    foregroundColor = backgroundColor.bodyTextColor;
```

This always has the right contrast I assume?

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 2]
```diff
@@ -59,6 +59,103 @@ function getContrastRatio(
...
+
+    // This color has the wrong contrast ratio, but it is the right color.
+    // Let's find similar colors that might have the right contrast ratio
+
+    const currentColor = colors[i];
+    if (DEBUG_COLOR) {
+      logColor(colors[i], "Finding similar color with better contrast");
+    }
+
+    for (let j = i + 1; j < colors.length; j++) {
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Little late now but Node-Vibrant has a DeltaE94 function we could probably use for this

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 3]
```diff
@@ -59,6 +59,103 @@ function getContrastRatio(
...
+      const diffScore =
+        Math.abs(currentColor.rgb[0] - compareColor.rgb[0]) +
+        Math.abs(currentColor.rgb[1] - compareColor.rgb[1]) +
+        Math.abs(currentColor.rgb[2] - compareColor.rgb[2]);
+
+      if (DEBUG_COLOR) {
+        logColor(colors[j], `${colors[j].hex} - ${diffScore}`);
+      }
+
+      if (diffScore > 150) {
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Why 150 ? Can me make this a CONST like Contrast Ratio

File: src/panels/lovelace/cards/hui-media-control-card.ts (hunk 2 above)
Comment (zsarnett) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: https://github.com/akfish/node-vibrant#deltae94lab1-number-lab2-number-number

Comment (balloob) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: Played with it. It doesn't seem to be a good match. If you look at the diff scores, there are colors considered "similar" but they have a complete different hue.

image

File: src/panels/lovelace/cards/hui-media-control-card.ts (hunk 3 above)
Comment (balloob) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: MAGIC.

**Review by zsarnett — APPROVED** [review]

**Pull request merged**